saleor/webhook/response_schemas.py

```diff
@@ -0,0 +1,79 @@
+import logging
+from decimal import Decimal
+from typing import Any, cast
+
+from graphql.error import GraphQLError
+from prices import Money
+from pydantic import BaseModel, Field, field_validator
+
+from ..app.models import App
+from ..core.enums import CurrencyEnum
+from ..core.utils.metadata_manager import method_metadata_is_valid
+from ..graphql.core.utils import from_global_id_or_error
+from ..shipping.interface import ShippingMethodData
+from ..shipping.models import ShippingMethod
+from .const import APP_ID_PREFIX
+from .transport.shipping_helpers import to_shipping_app_id
+
+logger = logging.getLogger(__name__)
+
+name_max_length = ShippingMethod._meta.get_field("name").max_length
+
+
+class ShippingMethodSchema(BaseModel):
+    id: str | int
+    name: str = Field(..., max_length=name_max_length)
+    amount: Decimal = Field(..., ge=0)
+    currency: CurrencyEnum
+    maximum_delivery_days: int | None = Field(None, ge=0)
+    minimum_delivery_days: int | None = Field(None, ge=0)
+    description: str | None = None
+    metadata: dict[str, str] | None = {}
+
+    @field_validator("metadata", mode="before")
+    @classmethod
+    def clean_metadata(cls, value: Any) -> dict[str, str]:
+        metadata = value
+        if metadata:
+            metadata = metadata if method_metadata_is_valid(metadata) else {}
+        return metadata or {}
+
+    @property
+    def price(self) -> Money:
+        return Money(self.amount, self.currency)
+
+    def get_shipping_method_data(self, app: "App"):
```

Comment:
question: why you use @property once and method second time?

I assume you need to pass "app" to the method which can't be done in property getter?

Is there any difference if you use `get_price()` instead?

(It's not a change request, I'm learning python now ;) )

Reply:
You can define `get_price()` as well. Only difference is that with property you can use it like a instance field `self.price` in case of method you need to call it to get the response `self.get_price()`